### PR TITLE
Adding fid calculation at the end of epochs (at train.py)

### PR DESCRIPTION
**Pull request by OmriKaduri**
Hi,

Thanks for a great repo for experimenting with CycleGAN and pix2pix! 

I wanted to compute FID scores during training (between the validation set and the generated images on validation), so I added this as an option at train.py.

Let me know if any changes needed. I believe it might be useful for others.

### CODE REVIEW

**Comment by rickkk856** [comment]
While we still don't have the implemented metrics I prepared simple way to get FID after training using test data _real images on domain B_ and _generated images from domain B_  --> input domain is simply ignored 

https://github.com/rickkk856/Batch_FID_PyTorchPix2Pix

**Comment by junyanz** [comment]
We cannot accept this PR at this time, due to two reasons: 
1. this PR (training time FID) may introduce too many changes to `train.py`. To add metrics to the training code, we need to add a metric abstract base class `metric.py` that can work for different metrics, implement an FID metric class `metric_FID.py`, and keep the training code simple. 
2. Pytorch-FID also has anti-aliasing issues.  FID calculation is quite complicated as it also involves which reference dataset you are using and how many samples you are using.  Please see the clean-fid [project](https://github.com/GaParmar/clean-fid) for more details. 

It might be great to first add the FID evaluation code for pre-trained checkpoints (after training is done). We will see if we could do that in the near future.
